src/executables/csql.c

```diff
@@ -1460,6 +1464,205 @@ csql_do_session_cmd (char *line_read, CSQL_ARGUMENT * csql_arg)
 	}
 
       break;
+
+    case S_CMD_CONNECT:
+      /*dbname_cpy can be stored DB name+ @ + remote hostname + \0 */
+      char dbname_cpy[MAXLOGNAME + CUB_MAXHOSTNAMELEN + 2];
+      char username_cpy[DB_MAX_USER_LENGTH + 1];
+      char host_cpy[DB_MAX_USER_LENGTH + 1];
+      const char *delim = " @\n";
+      char *save_ptr_strtok = NULL;
+      char *db_name = NULL;
+      char *user_name = NULL;
+      char *host_name = NULL;
+      char *p = NULL;
+      CSQL_ARGUMENT csql_new_arg;
+
+      /*find @hostname */
+      host_name = strchr (argument, '@');
+      if (host_name != NULL)
+	{
+	  if (*(host_name + 1) == '\0')
+	    {
+	      host_name = NULL;
+	      csql_Error_code = CSQL_ERR_CONNECT;
+	      return DO_CMD_FAILURE;
+	    }
+	  else
+	    {
+	      host_name += 1;
+	    }
+	}
+
+      if (host_name != NULL)
+	{
+	  if (strcasecmp ("127.0.0.1", host_name) == 0 || strcasecmp ("localhost", host_name) == 0
+	      || strcasecmp (boot_get_host_name (), host_name) == 0 || strcasecmp (boot_get_ip (), host_name) == 0)
+	    {
+	      *(host_name - 1) = '\0';
+	    }
+	}
```

Comment:
Do you have any reaon that removing hostname for localhost.

Reply:
- error case
;Connect public demodb@localhost

- if not remove,
call db_restart_ex()

Reply:
Could you check it again?
What you mean, db_restart_ex () do not accept dbname@host style?

Reply:
No, I mean when we login with locahost user like `;Connect public demodb@localhost`, function flow will go into the db_restart_ex () (line +3505)

Reply:
Users will be rather confused if the connection they try gives different results for local and remote. Personally I think it should yield the same result whether it is local or remote, even you could sustain current session for changing user on same database server. Whether or not the attempted connection fails or success, it would be good to clearly give the message that the current connection will be closed anyway.

Anyway, pls ask @mhoh3963 for it, I think it is a user requirements.

Reply:
I understand what you wrote comment.
From the my point of view, localhost is not necessary to call db_restart_ex as view of cost.

**But** from the user's point of view, as your suggest, I could add session end with this code.
`      if (csql_Database_connected)
        {
          csql_Database_connected = false;
          db_end_session ();
          db_shutdown ();
        }
`
And user always take same results.
Ok. I will ask him about our opinion. Thank you.